indigo_app/templates/indigo_api/work_popup.html

```diff
@@ -4,32 +4,55 @@ <h6>
     {% if work.stub %}
       <span class="badge badge-info">stub</span>
     {% endif %}
+    <br/><span class="text-muted mt-1">{{ work.frbr_uri }}</span>
   </h6>
-  <p class="text-muted mb-1 wb-all">{{ work.frbr_uri }}</p>
 
   {% if work.assent_date %}
-    <p class="mb-2">Assented to on {{ work.assent_date|date:"Y-m-d" }}.</p>
+    <p class="mb-2">Assented to on {{ work.assent_date|date:"Y-m-d" }}</p>
   {% endif %}
 
-  <p class="mb-2">Published on {{ work.publication_date|date:"Y-m-d" }}
-    {% if work.publication_name and work.publication_number and work.publication_document %}
-      in <a href="{% url 'work_publication_document' frbr_uri=work.frbr_uri filename=work.publication_document.filename %}" target="_blank" rel="noopener">
-        {{ work.publication_name }} no. {{ work.publication_number }}</a>.
-    {% elif work.publication_name and work.publication_number %}
-      in {{ work.publication_name }} no. {{ work.publication_number }}.
-    {% endif %}
-  </p>
+  {% if work.publication_name or work.publication_date %}
+    <p class="mb-2">
+      Published
+        {% if work.publication_date %}
+          on {{ work.publication_date|date:"Y-m-d" }}
+        {% endif %}
+
+        {% if work.publication_name %}
+          in
+
+          {% if work.publication_document %}
+            <a href="{% url 'work_publication_document' frbr_uri=work.frbr_uri filename=work.publication_document.filename %}" target="_blank" rel="noopener">{{ work.publication_name }}
```

Comment:
I suggest moving the content of the `a` tag into the block below it. Also line 27 over indented.